### DISCUSSION TITLE
Could a "change" event be useful?

### DISCUSSION THREAD

**Discussion by lgk-bsw** [LeaVerou/nudeui#8] [Ideas]
First I wanted to thank you for your amazing work for this repo, your code is a huge help for me to design my first Web Components.

I created for my project a simple `three-state-checkbox` and need a "change" event which triggers when the value state changes. 
The same could be useful for your [nd-rating](https://nudeui.com/nd-rating/) component. The callback function could give back the new value or like in `<input>` an Event object with a `target` property which contains the Element object of `<nd-rating>`.

I'm not really sure how the best practice with events in Web Components is. Registering a [CustomEvent](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent) probably would be great, so can add event listeners like this:

```js
document.querySelector("nd-rating").addEventListener("change", ({ target }) => {
    console.log("new value:" + target.value)
})
```

Or add a property `onchange`:

```js
document.querySelector("nd-rating").onchange = ({ target }) => {
    console.log("new value:" + target.value)
}
```

Or setting a HTML attribute:

```js
function myEventListenerFunction ({ target }) => {
    console.log("new value:" + target.value)
}
```
```html
<nd-rating onchange="myEventListenerFunction"></nd-rating>
```

**Comment by LeaVerou** [comment]
The `change` event [is an existing event of type `Event`](https://html.spec.whatwg.org/multipage/indices.html#event-change), so it would be inconsistent to use a `CustomEvent` for it. But yeah, it's generally good practice to have a `change` event as well, so that components can be handled the same as regular form elements.

**Reply by lgk-bsw** [reply]
Ah great. In my case I just dispatched the event inside the setter method:
```js
    set checked(_checked) {
        const oldChecked = this.#checked
        this.#checked = _checked
        this.#updateDom()

        if (oldChecked !== _checked) {
            this.dispatchEvent(new Event("change"))
        }
    }
```
